packages/headless/src/features/product-listing/product-listing-sort-slice.ts

```diff
@@ -0,0 +1,21 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {getProductListingSortInitialState} from './product-listing-sort-state';
+import {
+  registerProductListingSortCriterion,
+  updateProductListingSortCriterion,
+} from './product-listing-sort-actions';
+
+export const productListingSortReducer = createReducer(
+  getProductListingSortInitialState(),
+  (builder) => {
+    builder
+      .addCase(
+        registerProductListingSortCriterion,
+        (_, action) => action.payload
```

Comment:
I'm not sure I get it all, are you suggesting that we put something like `state.productListingSort || action.payload)` in the `register` action so that the initial state is kept? :thinking: 

If so, this is not how the `sort` slice is coded:

```
.addCase(registerSortCriterion, (_, action) =>
        buildCriterionExpression(action.payload)
      )
      .addCase(updateSortCriterion, (_, action) =>
        buildCriterionExpression(action.payload)
      )
```

Reply:
Indeed, I see it matches the `sort` slice. Aside from the action name though, there isn't a difference between the two actions, hmm.

For other slices, there's a defensive `if` check to prevent overwriting when a `register` action is fired.
Let's keep it as it is for now 🤞